pkg/asset/agent/manifests/nmstateconfig.go

```diff
@@ -124,15 +125,15 @@ func (n *NMStateConfig) Load(f asset.FileFetcher) (bool, error) {
 }
 
 func (n *NMStateConfig) finish() error {
+	if !common.AgentConfigAvaialble {
+		if n.Config == nil {
+			return errors.New("missing configuration or manifest file")
+		}
 
-	if n.Config == nil {
-		return errors.New("missing configuration or manifest file")
-	}
-
-	if err := n.validateNMStateConfig().ToAggregate(); err != nil {
-		return errors.Wrapf(err, "invalid NMStateConfig configuration")
+		if err := n.validateNMStateConfig().ToAggregate(); err != nil {
```

Comment:
I think that the `}` landed in the wrong place here; we shouldn't skip validation of NMStateConfig data just because there is a RendezvousIP defined.